data/src/client.rs

```diff
@@ -1225,12 +1242,32 @@ impl Client {
             Command::Numeric(RPL_ENDOFMONLIST, _) => {
                 return None;
             }
+            Command::MARKREAD(target, Some(timestamp)) => {
+                let timestamp = timestamp.strip_prefix("timestamp=").and_then(|timestamp| {
+                    DateTime::parse_from_rfc3339(timestamp)
+                        .ok()
+                        .map(|dt| dt.with_timezone(&Utc))
+                });
+                return Some(vec![Event::UpdateReadMarker(target.clone(), timestamp)]);
+            }
             _ => {}
         }
 
         Some(vec![Event::Single(message, self.nickname().to_owned())])
     }
 
+    pub fn send_markread(&mut self, target: &str, read_marker: Option<DateTime<Utc>>) {
+        if self.supports_read_marker {
+            if let Some(read_marker) = read_marker {
```

Comment:
This API should never be called if there's no read marker, since it doesn't do anything with None. Let's change the API to always take a value and handle the option at the call site.